ksqldb-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -306,20 +306,31 @@ static Expression rewriteTimeComparisonForFilter(final Expression expression) {
 
   @SuppressWarnings("unchecked")
   public SchemaKStream<Struct> selectKey(
+      final FormatInfo valueFormat,
       final Expression keyExpression,
       final Optional<FormatInfo> forceInternalKeyFormat,
-      final Stacker contextStacker
+      final Stacker contextStacker,
+      final boolean forceRepartition
   ) {
     final boolean keyFormatChange = forceInternalKeyFormat.isPresent()
         && !forceInternalKeyFormat.get().equals(keyFormat.getFormatInfo());
 
-    if (!keyFormatChange && repartitionNotNeeded(ImmutableList.of(keyExpression))) {
+    if (!keyFormatChange
+        && repartitionNotNeeded(ImmutableList.of(keyExpression))
+        && !forceRepartition
```

Comment:
"easy" meaning easy to grok or computationally easy to evaluate? (Already made the change, just curious about your motivations.)

Reply:
both, but mostly the computationally easy one. I don't know by reading this code that `repartitionNotNeeded` is something lightweight - it might call out to some external service for all I know without digging into it. If we put the simple boolean checks first I don't really need to care